src/main/src/main/java/org/geoserver/catalog/CoverageView.java

```diff
@@ -451,4 +507,72 @@ public boolean equals(Object obj) {
         if (selectedResolutionIndex != other.selectedResolutionIndex) return false;
         return true;
     }
+
+    /**
+     * Builds coverage bands for a given catalog store based on the specified composition type.
+     *
+     * @param catalog The catalog containing the coverage store
+     * @param storeId The ID of the coverage store to process
+     * @param compositionType The type of band composition to apply
+     * @param coverageBands List to populate with generated coverage band names
+     * @param inputCoverageBands Map to track input coverage band information
+     * @throws IOException If there is an error reading the coverage store
+     */
+    public static void buildCoverageBands(
+            Catalog catalog,
+            String storeId,
+            CompositionType compositionType,
+            List<String> coverageBands,
+            Map<String, Integer> inputCoverageBands)
+            throws IOException {
+        CoverageStoreInfo store = catalog.getStore(storeId, CoverageStoreInfo.class);
+        GridCoverage2DReader reader =
+                (GridCoverage2DReader) catalog.getResourcePool().getGridCoverageReader(store, null);
+        String[] coverageNames = reader.getGridCoverageNames();
+
+        for (String coverage : coverageNames) {
+            ImageLayout layout = reader.getImageLayout(coverage);
+            SampleModel sampleModel = layout.getSampleModel(null);
+            final int numBands = sampleModel.getNumBands();
+            if (CompositionType.JIFFLE.equals(compositionType)) {
+                inputCoverageBands.put(coverage, numBands);
+            }
+
+            switch (compositionType) {
+                case BAND_SELECT:
+                    if (numBands == 1) {
+                        // simple syntax for simple case
+                        coverageBands.add(coverage);
+                    } else {
+                        for (int i = 0; i < numBands; i++) {
+                            coverageBands.add(coverage + BAND_SEPARATOR + i);
+                        }
+                    }
+                    break;
+                case JIFFLE:
+                    String coverageBand = String.format(
```

Comment:
Do we really need to add all this information in the Jiffle band name? It also does not look like the GUI screenshots before were carring all this info in a name... have to figure out where all this is used.

Reply:
This is the summary reported for each coverage/band in the UI as part of the available inputs.
![image](https://github.com/user-attachments/assets/f90b79cd-27ad-46c7-8413-3f3653df17e6)